### PR TITLE
F.binary_cross_entropy_with_logits already applies a sigmoid

### PR DESCRIPTION
**Pull request by jbeaulieu-coveo**
The binary_cross_entropy_with_logits will already apply a sigmoid. Doing the operation before applies the sigmoid twice, which has the following impact

Since the first sigmoid bounds the output between 0 and 1, the second applies sigmoid([0,1]) = [0.5,0.73] , meaning the lowest score the model will see is 0.5

### CODE REVIEW

**Comment by itakurah** [comment]
> The binary_cross_entropy_with_logits will already apply a sigmoid. Doing the operation before applies the sigmoid twice, which has the following impact
> 
> Since the first sigmoid bounds the output between 0 and 1, the second applies sigmoid([0,1]) = [0.5,0.73] , meaning the lowest score the model will see is 0.5

Hi, can you please add a unit test to verify this logic?
